src/hidimstat/dcrt.py

```diff
@@ -11,265 +11,379 @@
 
 
 def _x_distillation_lasso(
-    X, idx, Sigma_X=None, cv=3, n_alphas=100, alpha=None, use_cv=False, n_jobs=1
+    X,
+    idx,
+    sigma_X=None,
+    cv=3,
+    alpha=None,
+    alpha_max_fraction=0.5,
+    n_alphas=100,
+    alphas=None,
+    n_jobs=1,
+    random_state=0,
 ):
     """
-    This function applies the distillation of the variable of interest with the
-    remaining variables using the lasso
+    Distill variable X[:, idx] using Lasso regression on remaining variables.
+
+    This function implements the distillation process to estimate the conditional
+    distribution of X[:, idx] given the remaining variables, using either Lasso
+    regression or a known covariance matrix.
+
+    Parameters
+    ----------
+    X : {array-like, sparse matrix} of shape (n_samples, n_features)
+        The input samples.
+    idx : int
+        Index of the variable to be distilled.
+    sigma_X : {array-like, sparse matrix} of shape (n_features, n_features), default=None
+        The covariance matrix of X. If provided, used instead of Lasso regression.
+    cv : int, cross-validation generator or iterable, default=3
+        Determines the cross-validation splitting strategy for LassoCV.
+    alpha : float, default=None
+        The regularization strength for Lasso. If None, determined automatically.
+    n_alphas : int, default=100
+        Number of alphas along the regularization path for LassoCV.
+    alphas : array-like, default=None
+        List of alphas where to compute the models.
+        If None alphas are set automatically.
+    alpha_max_fraction : float, default=0.5
+        Fraction of lambda_max to use when determining alpha.
+    n_jobs : int, default=1
+        Number of CPUs to use for cross-validation.
+    random_state : int, default=0
+        Random seed for reproducibility.
+
+    Returns
+    -------
+    X_res : ndarray of shape (n_samples,)
+        The residuals after distillation.
+    sigma2_X : float
+        The estimated variance of the residuals.
     """
     n_samples = X.shape[0]
     X_minus_idx = np.delete(np.copy(X), idx, 1)
 
-    if Sigma_X is None:
-        if use_cv:
-            clf = LassoCV(cv=cv, n_jobs=n_jobs, n_alphas=n_alphas, random_state=0)
-            clf.fit(X_minus_idx, X[:, idx])
-            alpha = clf.alpha_
-        else:
-            if alpha is None:
-                alpha = 0.1 * _alpha_max(X_minus_idx, X[:, idx])
-            clf = Lasso(alpha=alpha, fit_intercept=False)
-            clf.fit(X_minus_idx, X[:, idx])
+    if sigma_X is None:
+        # Distill X with least square loss
+        # configure Lasso and determine the alpha
+        clf, alpha = _fit_lasso(
+            X_minus_idx,
+            X[:, idx],
+            alpha=alpha,
+            alphas=alphas,
+            n_alphas=n_alphas,
+            cv=cv,
+            n_jobs=n_jobs,
+            random_state=random_state,
+            alpha_max_fraction=0.1,  # TODO: why 0.1 ?
```

Comment:
Do you have some justification for the value 0.1?

Reply:
Certainly no theoretical justification. I guess that it had decent empirical behavior in the experiments run initially.

Reply:
Ok, I will remove it.